src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(...args) {
+    if (args.length === 0) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += args.reduce((acc, num) => acc + num, 0);
```

Comment:
The use of `reduce` is redundant because the `adder` function is expected to receive only one argument at a time. You can simplify this by directly adding the argument to the `sum` variable.